target_selectors.go

```diff
@@ -0,0 +1,431 @@
+package wrapper
```

Comment:
Would target_selectors not belong in the command package/dir? what do you think?

Reply:
Not a bad spot for it. Its very much a WIP and I keep restructuring